### PR TITLE
add sbom generator task in the pipeline

### PR DESCRIPTION
**Pull request by arroyc**
https://azfunc.visualstudio.com/AzurePythonFx/_workitems/edit/1412

As title is mentioned .. 
### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -190,12 +190,32 @@ jobs:
...
+        echo "Generating V0 Integration Test Package for $BUILD_SOURCEBRANCHNAME"
+        LATEST_TAG=$(curl https://api.github.com/repos/Azure/azure-functions-python-worker/tags -s | jq '.[0].name' | sed 's/\"//g' |  cut -d'.' -f-2)
+        NUSPEC="pack\Microsoft.Azure.Functions.V0.PythonWorker.nuspec"
+        # Only required for Integration Test. Version number contains date (e.g. 3.1.2.20211028-dev)
+        WKVERSION="0.$LATEST_TAG-$(patchBuildNumberForDev)"
         echo "No Matching Release Tag For $BUILD_SOURCEBRANCH"
       fi
 
       echo "##vso[task.setvariable variable=nuspec_path]$NUSPEC"
       echo "##vso[task.setvariable variable=worker_version]$WKVERSION"
```
Comment (arroyc) [inline azure-pipelines.yml:204]: they are here temporarily to test the metadata generation task, will be commented out/removed once the testing is done

File: azure-pipelines.yml [hunk 2]
```diff
@@ -211,12 +231,12 @@ jobs:
...
-    displayName: '[Integration Test] Push NuGet package to the AzureFunctionsPreRelease feed'
+  #- task: NuGetCommand@2
+  #  condition: eq(variables['UPLOADPACKAGETOPRERELEASEFEED'], true)
+  #  inputs:
+  #    command: 'push'
+  #    packagesToPush: '$(Build.ArtifactStagingDirectory)/**/*.nupkg;!$(Build.ArtifactStagingDirectory)/**/*.symbols.nupkg'
+  #    nuGetFeedType: 'internal'
+  #    publishVstsFeed: 'e6a70c92-4128-439f-8012-382fe78d6396/f37f760c-aebd-443e-9714-ce725cd427df'
+  #    allowPackageConflicts: true
+  #  displayName: '[Integration Test] Push NuGet package to the AzureFunctionsPreRelease feed'
```
Comment (arroyc) [inline azure-pipelines.yml:242]: they are commented out temporarily to test the metadata generation task, will be uncommented out once the testing is done

File: azure-pipelines.yml [hunk 3]
```diff
@@ -164,7 +164,7 @@ jobs:
 
 - job: PackageWorkers
   dependsOn: ['Build_WINDOWS_X64', 'Build_WINDOWS_X86', 'Build_LINUX_X64', 'Build_OSX_X64']
-  condition: or(startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
+  #condition: or(startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
```
Comment (vrdmr) [inline azure-pipelines.yml:167]: We need this condition back. Right?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -190,7 +190,12 @@ jobs:
         # Only required for Integration Test. Version number contains date (e.g. 3.1.2.20211028-dev)
         WKVERSION="3.$LATEST_TAG-$(patchBuildNumberForDev)"
       else
-        echo "No Matching Release Tag For $BUILD_SOURCEBRANCH"
+        # line 194 to line 198 is for testing sbom generation
```
Comment (vrdmr) [inline azure-pipelines.yml:193]: Is this piece required?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -201,6 +206,28 @@ jobs:
...
+  - task: ManifestGeneratorTask@0
+    displayName: 'SBOM Generation Task'
+    inputs:
+      BuildDropPath: '$(Build.SourcesDirectory)'
+      BuildComponentPath: '$(Build.SourcesDirectory)'
+      Verbosity: 'Verbose'
+  - bash: |
+      # Write your commands here
+      
+        echo 'Hello world'
```
Comment (vrdmr) [inline azure-pipelines.yml:218]: As you have resolved this issue - can you please remove the echo statements?

**Review by vrdmr — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml (hunk 3 above)
Comment (arroyc) [reply azure-pipelines.yml:167]: yes, as per our discussion I commented this to make sure we have the manifest.json is getting created, I will uncomment this

File: azure-pipelines.yml (hunk 4 above)
Comment (arroyc) [reply azure-pipelines.yml:193]: nope this will be removed as well

File: azure-pipelines.yml [hunk 6]
```diff
@@ -201,6 +201,17 @@ jobs:
       buildType: 'current'
       downloadType: 'specific'
       downloadPath: '$(Build.SourcesDirectory)'
+  - task: ManifestGeneratorTask@0
+    displayName: 'SBOM Generation Task'
+    inputs:
+      BuildDropPath: '$(Build.SourcesDirectory)'
+      BuildComponentPath: '$(Build.SourcesDirectory)'
+      Verbosity: 'Verbose'
```
Comment (gavin-aguiar) [inline azure-pipelines.yml:209]: Why verbose? I think other languages workers are keeping it as 'Information'.
https://github.com/Azure/azure-functions-java-worker/pull/494/files

File: azure-pipelines.yml [hunk 7]
```diff
@@ -201,6 +201,17 @@ jobs:
...
+  - task: CopyFiles@2
+    inputs:
+      SourceFolder: '$(Build.SourcesDirectory)\_manifest'
+      Contents: '**'
```
Comment (gavin-aguiar) [inline azure-pipelines.yml:213]: Shouldn't contents be '*.*'?

**Comment by gavin-aguiar** [comment]
Are we not adding a validator task?

File: azure-pipelines.yml (hunk 7 above)
Comment (arroyc) [reply azure-pipelines.yml:213]: as per doc it says the following .. so I chose default case to make sure I'm not missing any subfolder contents
image

File: azure-pipelines.yml (hunk 6 above)
Comment (arroyc) [reply azure-pipelines.yml:209]: yes java chose information, but I chose it because I prefer more logging and it doesn't hurt.

**Comment by arroyc** [comment]
> Are we not adding a validator task?

not now

**Comment by gavin-aguiar** [comment]
> 

Got it. Can you update the PR title to remove the validator task

**Comment by gavin-aguiar** [comment]
> > Are we not adding a validator task?
> 
> not now

Got it. Can you update the PR title to remove the validator task?

File: azure-pipelines.yml [hunk 8]
```diff
@@ -201,6 +201,17 @@ jobs:
       buildType: 'current'
       downloadType: 'specific'
       downloadPath: '$(Build.SourcesDirectory)'
```
Comment (gavin-aguiar) [inline azure-pipelines.yml:203]: One suggestion, we can remove the Build.SourcesDirectory and change it to ArtifactStagingDirectory  from the download path in line 203. In this way, your sbom generation task will use Build.ArtifactStagingDirectory  and you can remove the CopyFiles task

Comment (arroyc) [reply azure-pipelines.yml:203]: good idea .. I will experiment with it .. I believe I tried it and it didn't work for some reason .. but I will try again. I merged my PR .. but I can send a new one to reflect that change

**Pull request merged**